lib/pulse/lib/Pulse.Lib.Array.Core.fsti

```diff
@@ -127,8 +127,10 @@ val gather
 
 val pts_to_range
   (#a:Type u#0)
-  ([@@@equate_strict] x:array a)
-  (i j : nat)
+  ([@@@mkey] x:array a)
+  ([@@@mkey] i : nat)
+  (j : nat)
+  (* ^NOTE: only using the start as matching key. *)
```

Comment:
The implementation file also adds the `[@@@mkey]` attribute to the end.

Reply:
Thanks. This reminds me of a possible footgun here: even inside `Array.Core.fst`, we will take the mkeys from the `val` and not the `let`. This is because the let is ascribed by the type in the `val`, and all attributes of the binders are basically overridden. Maybe F* should force binder attributes to be equal when subtyping function types.